api/versions.go

```diff
@@ -219,3 +221,43 @@ func (self *version1_2) HandleRequest(requestType string, request []string, m ma
 		return self.baseVersion.HandleRequest(requestType, request, m, w, r)
 	}
 }
+
+// v2.0 builds on v1.2
+type version2_0 struct {
+	baseVersion *version1_2
+}
+
+func newVersion2_0(v *version1_2) *version2_0 {
```

Comment:
I want to keep it based off 1.2 and replace whatever APIs we need to change. My guess is that some of the APIs will stay the same. If we rip out everything, we can remove the base dependency. 

I am fine going the other way and adding the base versions once we know that some of the endpoints will be replicated.